src/stepFunctions/commands/visualizeStateMachine/getStateMachineDefinitionFromCfnTemplate.ts

```diff
@@ -0,0 +1,68 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as fs from 'fs'
+import { getLogger, Logger } from '../../../shared/logger'
+
+/**
+ * @param {string} uniqueIdentifier - unique identifier of state machine
+ * @param {string} templatePath - path for the template.json file
+ * 
+ * @returns the escaped ASL Json definition string of the state machine construct
+ */
+export function getStateMachineDefinitionFromCfnTemplate(uniqueIdentifier: string, templatePath: string) {
+    const logger: Logger = getLogger()
+    try {
+        const data = fs.readFileSync(templatePath, 'utf8')
+        const jsonObj = JSON.parse(data)
+        const resources = jsonObj.Resources
+        let key
+
+        const matchingKeyList: string[] = []
+        for (const key of Object.keys(resources)) {
+            //the resources list always contains 'CDKMetadata'
+            if (key === 'CDKMetadata') continue
+
+            if (key.substring(0, uniqueIdentifier.length) === uniqueIdentifier) {
+                matchingKeyList.push(key)
+            }
+        }
+        if (matchingKeyList.length === 0) {
+            return
+        }
+        else if (matchingKeyList.length === 1) {
+            key = matchingKeyList[0]
+        }
+        else {
+            //return minimum length key in matchingKeyList
+            key = matchingKeyList.reduce((a, b) => a.length <= b.length ? a : b)
+        }
+
+        const definitionString = jsonObj.Resources[`${key}`].Properties.DefinitionString
+        return definitionString
+    }
+    catch (err) {
+        logger.debug('Unable to extract state machine definition string from template.json file.')
+        logger.error(err as Error)
+    }
+}
+
+interface CdkStateMachineWithPlaceholders {
+    "Fn::Join": (string | any)[]
```

Comment:
Can you get any more granular than `string | any`? At that point, you may as well just use `any` since you'll have to cast to `string` on the other side.